client/engine/engine.go

```diff
@@ -148,7 +149,10 @@ func (e *Engine) handleMessage(message protocols.Message) (ObjectiveChangeEvent,
 		if err != nil {
 			return ObjectiveChangeEvent{}, err
 		}
-
+		if objective.GetStatus() == protocols.Completed {
```

Comment:
I've introduced a new objective status `Completed`. It gets set by `Crank` when the protocol is completed. This let's us easily filter out objectives that are complete.